### PR TITLE
ST-152: New endpoint to return tax code changes

### PR DESCRIPTION
**Pull request by amaalali**
- [x] Unit tests passing
- [x] Coverage reached
- [x] Acceptance tests passing
- [x] Reviewed

### CODE REVIEW

File: app/uk/gov/hmrc/tai/service/TaxCodeChangeService.scala [hunk 1]
```diff
@@ -17,39 +17,83 @@
...
 
-  def taxCodeHistory(nino: Nino): Future[TaxCodeHistory] = {
-    val currentYear = TaxYear()
-    taxCodeChangeConnector.taxCodeHistory(nino, currentYear)
+  def taxCodeChange(nino: Nino): Future[TaxCodeChange] = {
+
+    taxCodeChangeConnector.taxCodeHistory(nino, TaxYear()) map { taxCodeHistory =>
+      val currentRecord :: previousRecord :: _ = sortedByDate(taxCodeHistory.operatedTaxCodeRecords)
+
+      val perviousEndDate = currentRecord.dateOfCalculation.minusDays(1)
```
Comment (SeanArmstrong) [inline app/uk/gov/hmrc/tai/service/TaxCodeChangeService.scala]: `perviousEndDate` vs `previousEndDate`

File: app/uk/gov/hmrc/tai/service/TaxCodeChangeService.scala [hunk 2]
```diff
@@ -17,39 +17,83 @@
...
+                                                      previousStartDate(previousRecord.dateOfCalculation),
+                                                      perviousEndDate,
+                                                      previousRecord.employerName)
+
+      TaxCodeChange(currentTaxCodeChange, previousTaxCodeChange)
+    }
   }
 
-}
+  private val inYearOf = (year: TaxYear) => (record: TaxCodeRecord) => TaxYear(record.dateOfCalculation) == year
```
Comment (SeanArmstrong) [inline app/uk/gov/hmrc/tai/service/TaxCodeChangeService.scala]: This is unused?

File: docs/taxaccount/tax-code-change.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
+Tax Code History
```
Comment (SeanArmstrong) [inline docs/taxaccount/tax-code-change.md]: Should this be Tax Code Change

File: docs/taxaccount/tax-code-change.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    }
+  },
+  "links" : [ ]
+}
+```
+
+* **Error Response:**
+
+  * **Code:** 401 UNAUTHORIZED <br />
+    **Content:** `{"statusCode":401,"message":"Authorisation refused for access to GET /tai/$NINO/tax-account/tax-code-history"}`
```
Comment (SeanArmstrong) [inline docs/taxaccount/tax-code-change.md]: this endpoint should be `/tax-code-change`?

File: docs/taxaccount/tax-code-changed.md [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+* **Success Response:**
+
+  * **Code:** 200 <br />
+    **Content:** Boolean
+
+* **Error Response:**
+
+  * **Code:** 401 UNAUTHORIZED <br />
+    **Content:** `{"statusCode":401,"message":"Authorisation refused for access to GET /tai/$NINO/tax-account/tax-code-change"}`
```
Comment (SeanArmstrong) [inline docs/taxaccount/tax-code-changed.md]: `/tax-code-changed`

File: project/MicroServiceBuild.scala [hunk 6]
```diff
@@ -15,8 +15,8 @@ object MicroServiceBuild extends Build with MicroService {
 
 private object AppDependencies {
   import play.sbt.PlayImport._
-  private val pegdownVersion = "1.4.2"
-  private val scalatestVersion = "2.2.2"
+  private val pegdownVersion = "1.6.0"
```
Comment (SeanArmstrong) [inline project/MicroServiceBuild.scala]: Have we spoke already about why these have changed? Are they relevant to the PR?

File: test/uk/gov/hmrc/tai/controllers/taxCodeChange/TaxCodeChangeRecordControllerSpec.scala [hunk 7]
```diff
@@ -87,28 +87,28 @@ class TaxCodeChangeControllerSpec extends PlaySpec with MockitoSugar with MockAu
...
-
-      when(mockTaxCodeService.taxCodeHistory(testNino)).thenReturn(Future.successful(taxCodeHistory))
-
-      val response = controller.taxCodeHistory(testNino)(FakeRequest())
-
-      contentAsJson(response) mustEqual Json.toJson(ApiResponse(taxCodeHistory, Nil))
-
-    }
-  }
+//  "taxCodeHistory" should {
```
Comment (SeanArmstrong) [inline test/uk/gov/hmrc/tai/controllers/taxCodeChange/TaxCodeChangeRecordControllerSpec.scala]: Remove or uncomment?

File: test/uk/gov/hmrc/tai/service/TaxCodeChangeServiceImplSpec.scala [hunk 8]
```diff
@@ -73,47 +139,136 @@ class TaxCodeChangeServiceImplSpec extends PlaySpec with MockitoSugar {
...
+
+  "taxCodeChange" should {
+
+    "return a domain TaxCodeRecord" when {
+
+      "there has been one daily tax code change in the year after annual coding" in {
+        val currentStartDate = TaxYearResolver.startOfCurrentTaxYear.plusDays(2)
+        val currentEndDate = TaxYearResolver.endOfCurrentTaxYear
+        val previousStartDate = TaxYearResolver.startOfCurrentTaxYear
+        val PreviousEndDate = currentStartDate.minusDays(1)
```
Comment (SeanArmstrong) [inline test/uk/gov/hmrc/tai/service/TaxCodeChangeServiceImplSpec.scala]: [minor] capitalised val. Was this for a specific reason?

File: test/uk/gov/hmrc/tai/service/TaxCodeChangeServiceSpec.scala [hunk 9]
```diff
@@ -1,98 +0,0 @@
-/*
```
Comment (SeanArmstrong) [inline test/uk/gov/hmrc/tai/service/TaxCodeChangeServiceSpec.scala]: Was this an intentional delete?

Where is the specs for `TaxCodeChangeService.scala`

File: project/MicroServiceBuild.scala (hunk 6 above)
Comment (amaalali) [reply project/MicroServiceBuild.scala]: To sync with Tai-Frontend

**Pull request merged**